File: fetch-items.ts


const WFM_BASE_URL = "https://api.warframe.market/v2";
const HEADERS = { 'Platform': 'pc', 'Language': 'ru' };

async function fetchAndSaveItems() {
  console.log("Запускаю скрипт для получения списка всех предметов с Warframe Market...");
  
  try {
    const response = await fetch(`${WFM_BASE_URL}/items`, { headers: HEADERS });
    if (!response.ok) {
      throw new Error(`HTTP ${response.status}: ${response.statusText}`);
    }
    
    const data = await response.json();
    const items = data.data || [];
    
    // Создаем папку 'data', если ее нет
    const dataDir = path.join(process.cwd(), 'data');
    if (!fs.existsSync(dataDir)) {
      fs.mkdirSync(dataDir);
    }
    
    // Сохраняем данные в файл
    const filePath = path.join(dataDir, 'items.json');
    fs.writeFileSync(filePath, JSON.stringify(items, null, 2));
    
    console.log(`✅ Успешно! Список из ${items.length} предметов сохранен в файл: ${filePath}`);
    
  } catch (error) {
    console.error('❌ Ошибка при получении или сохранении списка предметов:', error);
  }
}

fetchAndSaveItems();

File: fetch_items.py
import json
import os
import requests # Возможно, понадобится установить: pip install requests

# Константы, как в вашем коде
WFM_BASE_URL = "https://api.warframe.market/v2"
HEADERS = {
    'Platform': 'pc',
    'Language': 'ru',
    'User-Agent': 'Warframe-Inventory-Fetcher/Python-v1'
}

def fetch_and_save_items():
    """
    Скачивает список всех предметов с Warframe Market
    и сохраняет его в файл data/items.json
    """
    print("Запускаю скрипт для получения списка всех предметов с Warframe Market...")
    
    try:
        # 1. Делаем запрос к API
        response = requests.get(f"{WFM_BASE_URL}/items", headers=HEADERS)
        response.raise_for_status()  # Вызовет ошибку, если статус не 200 OK
        
        data = response.json()
        items = data.get("data", [])
        
        # 2. Создаем папку 'data', если ее нет
        data_dir = os.path.join(os.getcwd(), 'data')
        if not os.path.exists(data_dir):
            os.makedirs(data_dir)
        
        # 3. Сохраняем данные в файл
        file_path = os.path.join(data_dir, 'items.json')
        
        with open(file_path, 'w', encoding='utf-8') as f:
            # indent=2 для красивого форматирования, ensure_ascii=False для кириллицы
            json.dump(items, f, indent=2, ensure_ascii=False)
            
        print(f"✅ Успешно! Список из {len(items)} предметов сохранен в файл: {file_path}")

    except requests.exceptions.RequestException as e:
        print(f"❌ Ошибка сети при запросе к API: {e}")
    except Exception as e:
        print(f"❌ Произошла непредвиденная ошибка: {e}")


# Запускаем нашу функцию
if __name__ == "__main__":
    fetch_and_save_items()